chain/xlm/client/client.go

```diff
@@ -0,0 +1,471 @@
+package client
+
+import (
+	"context"
+	"math"
+	"strconv"
+
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"time"
+
+	xc "github.com/cordialsys/crosschain"
+	xcbuilder "github.com/cordialsys/crosschain/builder"
+	"github.com/sirupsen/logrus"
+
+	"github.com/cordialsys/crosschain/chain/xlm/client/types"
+	xlminput "github.com/cordialsys/crosschain/chain/xlm/tx_input"
+	xclient "github.com/cordialsys/crosschain/client"
+	"github.com/stellar/go/xdr"
+)
+
+type Client struct {
+	Url        string
+	HttpClient *http.Client
+	Asset      xc.ITask
+	Passphrase string
+}
+
+var _ xclient.FullClient = &Client{}
+var _ xclient.ClientWithDecimals = &Client{}
+
+func NewClient(cfgI xc.ITask) (*Client, error) {
+	cfg := cfgI.GetChain()
+	networkPassphrase := cfg.ChainIDStr
+	if networkPassphrase == "" {
+		error := "Stellar configuration is missing ChainIDStr."
+		logrus.Error(error)
+		return nil, errors.New(error)
+	}
+
+	if cfg.ChainGasPriceDefault < 0 {
+		return nil, fmt.Errorf("ChainGasPriceDefault cannot be negative: %f", cfg.ChainGasPriceDefault)
+	}
+
+	if cfg.ChainMinGasPrice < 0 {
+		return nil, fmt.Errorf("ChainMinGasPrice cannot be negative: %f", cfg.ChainMinGasPrice)
+	}
+
+	return &Client{
+		Url:        cfg.URL,
+		HttpClient: http.DefaultClient,
+		Asset:      cfgI,
+	}, nil
+}
+
+// FetchTransferInput returns tx input for a Template tx
+func (client *Client) FetchTransferInput(ctx context.Context, args xcbuilder.TransferArgs) (xc.TxInput, error) {
+	config := client.Asset.GetChain()
+	txInput := xlminput.NewTxInput(config.ChainIDStr)
+	account := args.GetFrom()
+	currentSequence, err := client.FetchSequenceNumber(account)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch account sequence number: %w", err)
+	}
+
+	ledger, err := client.FetchLatestLedgerInfo()
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch ledger info: %w", err)
+	}
+
+	txInput.Sequence = currentSequence + 1
+	txInput.MinLedgerSequence = ledger.Sequence
+
+	defaultFee := config.ChainGasPriceDefault
```

Comment:
Can we estimate this dynamically?  It will be too tedious to make config updates to keep up with XLM fee market.